### ISSUE TITLE
Clarify to users how locations are handled when recording surveys

### ISSUE THREAD

**Issue by kitenetter** [NERC-CEH/irecord-app#263]
It appears that we somewhere between 3,000 and 6,000 app records that are mislocated, due to users not realising how the app deals with geolocations for individual records that are part of a survey. The most likely cause of this stems from users setting up a survey when out in the field, and then continuing to add further records when they have moved away from the survey area (e.g. upon returning home).

We need to make it clearer to users how the app is behaving when adding records to a survey. Suggestions:
1. Ensure that the default setting is for individual records not to be geolocated, i.e. the default position is that they all get linked to the location set up for the survey
2. Add an information box to the survey data entry page so that users are told something like "All records added to this survey will be assigned to the location that you have set up for the survey. If you want to assign precise locations to records within the survey, make sure that you **switch on** the option to "Geolocate survey entries". [Ideally users should be able to do this from the page itself, without having to go to the app settings.]
3. If the option to "Geolocate survey entries" has been switched on, the survey page should inform users "All records added to this survey will be given individual locations, e.g. based on the current GPS location for your device. If you want to add records after you have moved away from the survey area (e.g. after returning home) please **switch off** the option to "Geolocate survey entries".

Other possibilities we could consider: enabling some sort of calculated warning so that if a user is more than, say, 1 or 2 kms from their survey location they get a warning? Not sure how feasible that is.

Enable a map display so that users can see all the records from an individual survey before they upload them?

Other ideas welcome!

**Comment by kazlauskis** [comment]
Thanks, this could be clearer to the user, and I can do all those options. It would be easy for me to switch off the individual records geolocation default setting. 

These misplaced 3-6k records are around 4-8% of the total survey uploads, and the remaining 92% of occurrences have accurate locations. It _might_ be more beneficial to continue capturing precise individual record locations and address the problem of recording from home instead. So I particularly like your last two suggestions - detecting the distances and showing occurrences on the map. We do similar things in other apps: for the iRecord Butterflies app, we show warnings for out-of-UK records; for the ButterflyCount, a warning if the user records a very long walked distance during 15min counts. It would be easy to detect a large distance between the parent sample location and an occurrence and show a warning. Also, we could show all the occurrences on the parent sample map (we do this for the ButterflyCount app). It might be helpful to add this functionality to the map anyway because showing all occurrences of a survey would indicate if any are outside the parent grid square (i.e. monad) if users want to do grid recording.

**Comment by kazlauskis** [comment]
@kitenetter I have now done the calculated warning for species entries that are over 5km away from the survey location and also added species entries to the survey map. Does that look OK? Do you think this will be enough to clarify how the child sub-samples are located, or should I also implement your 1st suggestion - switch off GPS locations for list entries by default?
<img width="411" alt="Screenshot 2023-11-10 at 13 37 45" src="https://github.com/NERC-CEH/irecord-app/assets/7383162/9c736227-defd-491f-93ad-e04aeb3ebd5f">
<img width="398" alt="Screenshot 2023-11-10 at 13 35 25" src="https://github.com/NERC-CEH/irecord-app/assets/7383162/1df159bd-86e4-4bf5-9da8-282bf2f9be05">

**Comment by kitenetter** [comment]
Hi @kazlauskis this issue has come up again, and apologies for not having responded to your comments above. The changes you've made are very welcome, but I don't think they go far enough.

The warning message about distant locations (screenshot below) is fine, but is easy to miss - I think it needs to be larger and/or outlined in red or something so that users cannot miss it before they upload their records. We really do need to prevent the mislocation errors before they get uploaded.

The distance of 5 km for the warning seems too large to me. I think there is an argument for setting the warning so that it appears for any record that is outside the square chosen for the main survey, but if that seems too restrictive then I would suggest a warning for anything that is over 1 km from the main location.

The display of the individual records as green dots on the map is also helpful but also easy to overlook. If I look at the map for the survey as a whole it zooms in on the survey grid ref and I cannot see any outlying records unless I zoom out. And if if I look at the map for an outlying record it zooms to that record so it is not very obvious how far the record is from the main survey location.

So I think we do need to turn off the individual record geolocation by default. Ideally it would be good to have a link on the survey data entry page that would enable the individual geolocation to be turned back on from within that page, instead of having to come out and go in to the settings to find it.

**Comment by japonicus** [comment]
Is it possible to estimate what proportion of the errors are due to people straying outside the survey bounds while surveying in the field _v's_ spuriously GPS'ed entries added later at home?

Individual precise locations for each occurrence are valuable and are one of the major strengths of app-based recording, at least for taxon groups with fixed locations such as plants and fungi - perhaps less so for animals.

If many of the errors are due to straying in the field, then turning off GPS merely hides the problem and results in incorrectly located records that are difficult to detect later.

An alternative might be to make GPS tracking easy to toggle and to stop tracking following an unexpectedly large temporal/spatial jump between records - e.g. when someone drives home. (the prototype BSBI recording app turns off tracking if someone moves too quickly - because we anticipate that surveys are conducted on foot)

**Comment by kitenetter** [comment]
The serious errors that have been identified appear to relate to people adding data when at home or from widely differing localities. I take your point about the value of precise locations within a survey, and that we don't want to lose that functionality. But at the moment we are getting records verified in entirely the wrong locations.

Perhaps if we can make the warning message clearer we won't need to turn off the default geolocation.

We do need to enable people to add records to a field survey after they have returned home, so making it easier for the user to be aware of the geolocation setting and turn it off when required is definitely desirable.

**Comment by kazlauskis** [comment]
@kitenetter, would these changes help?

- [ ] Reduce 5km to 1km. The sub-sample distance is calculated from the parent survey location.
- [ ] Make the warning message stand out.
- [ ] Add a survey warning symbol to the pending surveys list page. This would make such records stand out from other pending records before upload.

Pending records in the app have two states, "Draft" and "Finished". Both can be modified before upload, so what if we turn off the automatic GPS for finished records? Essentially, users would compile their species lists in the field, press the Finish button in the header and then come back home, where they can add more species assigned to the survey location before the upload.

**Comment by kitenetter** [comment]
@kazlauskis I think your first two suggestions are definitely worth doing. I'm not sure I understand the third one - would the warning symbol be shown against all surveys, or just surveys that include some records more than 1 km from the parent location? But in any case I think it is worth trying.

I don't have any feel for whether recorders recognise the distinction between the "Draft" and "Finished" stages of a survey, so I'm not sure how well these states match to "in the field" and "back at home". I can see that turning off the auto GPS for finished surveys might be helpful, but it will depend on recorder behaviour. If it can be done in such a way that the recorder is clearly informed of what is happening, and can easily see how to switch the GPS back on if they need to, it might be worth trying but without some testing it's hard to predict whether recorders will find it helpful.

So I agree that we should implement your three tick-box options, and on balance I think we should try implementing the auto switch-off for "Finished" surveys, although it needs to be kept under review.

**Comment by kitenetter** [comment]
@kazlauskis I've just had another example of a user who is very upset that their records are located in the 'wrong place', due to having done data entry at home after starting a survey in the field. But in this case their field site is less than a kilometre from their home site. The difference in distance is not great, but the result is that records are being located in their garden when they intended them to be from a nearby natural habitat.

So although the measures we've discussed above are helpful, they wouldn't resolve things for this type of example. I think we really need to find a way of making it very obvious to the user, at the point of them entering an individual occurrence, whether or not it is being auto-located, and where the auto-location is in relation to the survey.

Whenever the user has the GPS switched on and is adding records to a survey, can we provide a clear warning to each individual record being added that says "This record is being added at your current GPS location - please edit the location if you have moved away from your survey area".

**Comment by kitenetter** [comment]
I also think that when someone starts a new survey on the app that we need to give them a very clear message about whether or not the GPS auto-location is switched on, with a reminder that records added after moving away will need a manual grid ref adding, and ideally give them the option of switching the GPS on or off at the point of data entry.

**Comment by kitenetter** [comment]
Apologies @kazlauskis - having done some more investigation it appears that the user mentioned in my previous two comments was NOT using the survey option, they were adding individual records, They were getting confused over the auto-location and user-specified location controls, and also over the field locking options.

It is of course possible that other users may be using the surveys and ending up with records that are near the survey location but not actually located correctly. And I still think it would be helpful for users to be given clearer information about whether their records are being auto-located or not at the point of data entry, if we can find a way of doing that.

**Comment by kazlauskis** [comment]
I agree, we can clarify how this works. I just don't want to affect the majority of the users if they don't have this problem. Most similar apps (like iNaturalist, iObs, eBird) will automatically geolocate users' records and will only show the resulting lat/long at the survey level without any warnings. The main reason is that these apps are designed for entering records in the field and most people will use a computer to enter records at home. Having said that, users can use the iRecord App at home a) for individual records they can lock their location to whatever place and the GPS won't be used b) for species list entries there is the app setting to disable the GPS for species and only use the survey location c) they can switch off the device location services to disable the GPS altogether.  I suggest we add a help text section to the Help page to explain how the geolocation works. 

Also, what if we also show a little location preview, something like this? 

![Screenshot 2024-01-18 at 12 01 52](https://github.com/NERC-CEH/irecord-app/assets/7383162/b284c54a-55fc-42fc-bdc3-73c1ef7d4965)

**Comment by sacrevert** [comment]
If this can be toggled by survey, I would prefer Martin's option 1 for the Plant Survey option, that is, that the record level geolocation defaults to off. Of course I agree that very precise records are useful, but this is less true for common plants, and I think the errors and user dissatisfaction outweighs the nicety of being able to say everything is located to GPS precision tbh.

**Comment by sacrevert** [comment]
Perhaps the simplest option is just to do what is mentioned a few times above: just put the geolocate option on the front page of every survey (in addition to whatever warnings are deemed appropriate across or within survey types).

I think the other warnings will always lead to errors , as it won't be that unusual to have a lunch/dinner near to the survey site whereupon records are entered. And presumably tracking whether people are stationary is very variably efficient too (often the first however many minutes of grid square or site plant survey are spent near-stationary ticking off all the common taxa one can see without moving).

**Comment by kazlauskis** [comment]
I have made it so that sub-samples are not automatically geolocated. There is a new toggle option that, if disabled, will not geolocate species entries when entered. Users can still manually set or geolocate individual entries when this option is off.

It would be helpful if someone could tackle https://github.com/Indicia-Team/warehouse/issues/533 as part of this update.

![Image](https://github.com/user-attachments/assets/0b7c5f06-eaa7-47bd-92fd-7c287934049a)

**Comment by sacrevert** [comment]
Looks good. What does this mean for precision information (presumably `@spatialRefPrecisionAttrId`)? Does it get overwritten/filled in appropriately, and is the attribute ID stored in that field correct for all survey types? For example, I know this is sample attribute ID `1110` (GPS Precision) for bryophyte survey when data are entered through the webpage (https://irecord.org.uk/enter_bryophyte_records). Perhaps linked to #302 ?

**Comment by kazlauskis** [comment]
@sacrevert if a location isn't attached to the sub-sample, at the moment, the sub-sample layer is dropped and only an occurrence is saved in the top survey sample. The idea in the https://github.com/Indicia-Team/warehouse/issues/533 is that we would still submit sub-samples but inherit the top survey values. We haven't discussed location accuracy values, but maybe these could be standardised as a core Indicia attribute and this way copied to sub-samples too.

**Comment by kitenetter** [comment]
@kazlauskis is there any more we need to resolve prior to the next app release?

Does https://github.com/Indicia-Team/warehouse/issues/533 need to be tackled before the release?

**Comment by kazlauskis** [comment]
@kitenetter yes, my part for this ticket is done, but we still need to tweak the warehouse.

**Comment by johnvanbreda** [comment]
The warehouse has been "tweaked" as per https://github.com/Indicia-Team/warehouse/issues/533.
